### PR TITLE
DAT640 2022 content

### PR DESCRIPTION
**Pull request by WerLaj**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,61 @@
...
+This repository contains course material (lectures, labs, and assignments) for the 2022 fall edition of the course.
+
+## Course format
+
+The course followed a standard format, with lectures, assignments and group projects. Physical lecturing led by the lecturer, included discussions, Kahoot quizzes, and hands-on exercises. The lectures were not be recorded/streamed, physical attendance was expected. Individual assignments  led by the TAs were dedicated time to work on graded assignments. Student that needed help with assignments were expected to attend in person. Group projects meetings were 15 minutes time slots dedicated for getting help/feedback on progress and ideas.
+
+The semester (consisting of 13 weeks) was divided into three periods, with two [lectures](lectures/) per week and [individual assignments](assignments/) during the first three weeks, one lecture per week, individual assignments and [group project work](projects/) during the next seven weeks, and last three weeks dedicated entirely for group projects.
+Dedicated time was provided to get help with the individual assignments (from TA) as well as to get feedback on the group project (from lecturer and TA).
```
Comment (kbalog) [inline README.md:10]: Two periods: (1) lectures and individual assignments during the first X weeks, with group projects running in parallel, (2) last 3 weeks dedicated entirely to the group projects.

Comment (kbalog) [inline README.md:10]: TAs

File: README.md [hunk 2]
```diff
@@ -1,3 +1,61 @@
...
+| 9 |  |  |  |
+| 10 |  |  |  |
+| 11 |  |  |  |
+| 12 |  |  |  |
+| 13 |  |  |  |
+| 14 |  |  |  |
+
+## Assignments
+
+| **Id** | **Topic** | **Resources** |
```
Comment (kbalog) [inline README.md:41]: Resource => Points

File: README.md [hunk 3]
```diff
@@ -1,3 +1,61 @@
...
+| 8 |  |  |  |
+| 9 |  |  |  |
+| 10 |  |  |  |
+| 11 |  |  |  |
+| 12 |  |  |  |
+| 13 |  |  |  |
+| 14 |  |  |  |
+
+## Assignments
+
```
Comment (kbalog) [inline README.md:40]: Need a single-paragraph explanation of how assignments work: code skeletion is to be completed, around 10 days to do the task, public and private tests...

File: README.md [hunk 4]
```diff
@@ -1,3 +1,61 @@
...
+| 10 |  |  |
+
+### Textbooks
+
+  * [Zhai & Massung. Text Data Management and Analysis: A Practical Introduction to Information Retrieval and Text Mining.  Morgan & Claypool, 2016.](https://dl.acm.org/doi/book/10.1145/2915031)
+  * [Balog. Entity-Oriented Search. Springer, 2018.](https://eos-book.org/)
+
+## Contributors
+
+Krisztian Balog (lecturer), Ivica Kostric (lecturer), Nolwenn Bernard (TA), Weronika Lajewska (TA)
```
Comment (kbalog) [inline README.md:61]: KB: lecturer => course responsible

**Review by kbalog — CHANGES_REQUESTED** [review]
- Remove assignments
- Add all slides under lectures and just link them directly from the main README (no need for module-specific folders)

File: README.md (hunk 1 above)
Comment (WerLaj) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (WerLaj) [reply README.md]: Done

File: README.md (hunk 3 above)
Comment (WerLaj) [reply README.md:39]: Done

File: README.md (hunk 4 above)
Comment (WerLaj) [reply README.md]: Done

**Comment by WerLaj** [comment]
I updated points for projects so that all deliveries are taken into consideration.

File: README.md [hunk 5]
```diff
@@ -1,3 +1,61 @@
-# DAT640-2022: Information Retrieval and Text Mining course at UiS
+# DAT640-2022: Information Retrieval and Text Mining course, University of Stavanger
 
-Materials for public release (i.e., links to slides and exercises) will be shared in this repository _after_ the semester.
+This repository contains course material (lectures, labs, and assignments) for the 2022 fall edition of the course.
+
+## Course format
+
+The course followed a standard format, with lectures, assignments and group projects. Physical lecturing led by the lecturer, included discussions, Kahoot quizzes, and hands-on exercises. The lectures were not be recorded/streamed, physical attendance was expected. Individual assignments  led by the TAs were dedicated time to work on graded assignments. Student that needed help with assignments were expected to attend in person. Group projects meetings were 15 minutes time slots dedicated for getting help/feedback on progress and ideas.
```
Comment (kbalog) [inline README.md:7]: - "were not be recorded/..." => "were not recorded/..."
- replace "Individual assignments ... " sentence => "Labs were led by TAs and dedicated to working on individual (graded) assignments."
- replace "Group projects meetings ..." sentence => "For the group projects, groups could book 15min weekly slots to get help/feedback on progress and ideas from the lecturer/TAs."

File: README.md [hunk 6]
```diff
@@ -1,3 +1,61 @@
...
+| 2 | Text classification and similarity | [lecture](lectures/M2-text_classification_and_clustering.pdf) | [exercises](exercises/E2) |
+| 3 | Search engine architecture and basic retrieval models | [lecture](lectures/M3-se_architecture_and_retrieval_models.pdf) | [exercises](exercises/E3) |
+| 4 | Advanced retrieval models | [lecture](lectures/M4-advanced_retrieval_models.pdf) | [exercises](exercises/E4) |
+| 5 | Retrieval evaluation | [lecture](lectures/M5-retrieval_evaluation.pdf) | [exercises](exercises/E5) |
+| 6 | Knowledge bases and entity retrieval | [lecture](lectures/M6-knowledge_bases_and_entity_retrieval.pdf) | [exercises](exercises/E6) |
+| 7 | Entity linking | [lecture](lectures/M7-entity_linking.pdf) | [exercises](exercises/E7) |
+| 8 | Semantic search | [lecture](lectures/M8-eos.pdf) | [exercises](exercises/E8) |
+| 9 | Word embeddings and dense retrieval | [lecture](lectures/M9-neural-ir-1.pdf) | [exercises](exercises/E9) |
+| 10 | Transformer-based models | [lecture](lectures/M10-neural-ir-2.pdf) | [exercises](exercises/E10) |
+| 11 | Conversational Information Access | [lecture](lectures/M11-CIA.pdf) |  |
```
Comment (kbalog) [inline README.md:33]: Capitalize consistently: Conversational information access

File: README.md [hunk 7]
```diff
@@ -1,3 +1,61 @@
...
+| 12 | Advanced evaluation | [lecture](lectures/M12-advanced_evaluation.pdf) |  |
+| 13 | Conversational Search Systems | [lecture](lectures/M13-css.pdf) | [exercises](exercises/E13) |
```
Comment (kbalog) [inline README.md:35]: Capitalize consistently: Conversational search systems

File: README.md [hunk 8]
```diff
@@ -1,3 +1,61 @@
...
+| 14 | Fairness and transparency in IR | [lecture](lectures/M14-fairness_transparency_ir.pdf) | [exercises](exercises/E14) |
+
+## Assignments
+
+During the course, students had to complete 9 assignments (deliverables) individually. These were graded and accounted for 50% of the project work (i.e., 20% of the final grade). Students were given a certain time limit (typically 10 days) to complete each assignment. For each assignment students were provided with the Python file with code skeleton which they were expected to complete according to the task description. Assignments were graded automatically, using a combination of public and hidden tests. Public tests were released together with the assignment; if the solution passed these tests, it was likely correct. However, in addition to those tests, there were some hidden tests that the solution was tested against after submission; these typically contained larger inputs/datasets, corner cases, or other inputs in order to test that the student fully understood the methods and/or followed the instructions.
```
Comment (kbalog) [inline README.md:40]: - "with the Python file" => "with a Python file"
- "code skeleton which" => code skeleton, which"

File: README.md [hunk 9]
```diff
@@ -1,3 +1,61 @@
...
+
+| **Id** | **Topic** | **Points** |
```
Comment (kbalog) [inline README.md:42]: Add this above the table: "The assignments are not released publicly as they might be reused in future editions of the course. (Feel free to reach out to us in email in you would like to get a copy of these.)"

File: README.md [hunk 10]
```diff
@@ -1,3 +1,61 @@
...
+| -- | -- | -- |
+| 1 | Classifier: Feature extraction | 2 |
+| 2 | Classifier: Evaluation | 3 |
+| 3 | Classifier: Training and evaluation | 5 |
+| 4 | Search engine: Indexing | 4 |
+| 5 | Search engine: Scoring | 7 |
+| 6 | Search engine: Evaluation | 4 |
+| 7 | Entity retrieval | 8 |
+| 8 | Entity ranking | 7 |
+| 9 | Learning to Rank | 10 |
```
Comment (kbalog) [inline README.md:52]: Use the labels used in Canvas: A1.1, A1.2, ... A3.3

File: projects/README.md [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+    - [Discussion group (on Canvas)](https://stavanger.instructure.com/courses/8838/discussion_topics/120142)
+  * **[Semantic Answer Type Prediction 2020](https://smart-task.github.io/2020/)**
+    - Given a question in natural language, the task is to predict type of the answer using a set of candidates from a target ontology (DBpedia).
+    - Document collection: DBpedia ([dump 2016-10](http://downloads.dbpedia.org/2016-10/)) and/or Wikidata ([dump](https://dumps.wikimedia.org/wikidatawiki/entities/))
+    - Resources
+      - [Task GitHub repo](https://github.com/smart-task/smart-dataset)
+      - Papers: [Balog & Neumayer (2012)](https://krisztianbalog.com/files/cikm2012-querytypes.pdf), [Garigliotti et al. (2017)](https://krisztianbalog.com/files/sigir2017-qt.pdf), [Setty & Balog (2020)](https://krisztianbalog.com/files/iswc2020-smart.pdf)
+    - [Discussion group (on Canvas)](https://stavanger.instructure.com/courses/8838/discussion_topics/120145)
+
+
```
Comment (kbalog) [inline projects/README.md:36]: Add `## Instructions`, and move all headers below a level down, i.e., `##` => `###`, `###` => `####`, etc.

File: projects/README.md [hunk 12]
```diff
@@ -0,0 +1,101 @@
+# Group project
+
+## Objectives
```
Comment (kbalog) [inline projects/README.md:3]: Remove heading

File: projects/README.md [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+
+The objective of project work is to apply the knowledge gained in the course, in a group setting, to a selected (open) information retrieval problem.
```
Comment (kbalog) [inline projects/README.md:5]: "project work is to apply" => "project work was to apply"

**Review by kbalog — CHANGES_REQUESTED** [review]
Nice work! A few comments added. 

Make sure you add empty lines above/below heading, tables, etc. when making changes.

File: README.md (hunk 5 above)
Comment (WerLaj) [reply README.md]: Done

File: README.md (hunk 6 above)
Comment (WerLaj) [reply README.md]: Done

File: README.md (hunk 7 above)
Comment (WerLaj) [reply README.md]: Done

File: README.md (hunk 8 above)
Comment (WerLaj) [reply README.md]: Done

File: README.md (hunk 9 above)
Comment (WerLaj) [reply README.md:44]: Done

File: README.md (hunk 10 above)
Comment (WerLaj) [reply README.md]: Done

File: projects/README.md (hunk 12 above)
Comment (WerLaj) [reply projects/README.md]: Done

File: projects/README.md (hunk 13 above)
Comment (WerLaj) [reply projects/README.md]: Done

**Review by kbalog — APPROVED** [review]
LGTM
